### PR TITLE
CLN: some code cleanups

### PR DESCRIPTION
**Pull request by ShaharNaveh**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/_libs/interval.pyx
Comment (jbrockmendel) [inline pandas/_libs/interval.pyx]: its worth double-checking me on this, but i think for non-cdef/cpdef functions, using cython syntax for the annotations doesnt help perf.  in these cases, let's just use python versions

File: pandas/_libs/tslib.pyx
Comment (jbrockmendel) [inline pandas/_libs/tslib.pyx]: do we usually use dashes or asterisks or something for this kind of bullet point thing?

File: pandas/_libs/tslib.pyx
Comment (jbrockmendel) [inline pandas/_libs/tslib.pyx]: dtype is always object, the type contained is specified by box.  let's also specify ndarray instead of just array

File: pandas/_libs/tslib.pyx
Comment (jbrockmendel) [inline pandas/_libs/tslib.pyx]: possibly can make `const int64_t[:]`?

File: pandas/_libs/internals.pyx
Comment (jbrockmendel) [inline pandas/_libs/internals.pyx]: some recent profiling has made me think this might be worth trying to optimize even harder.  two ideas come to mind:

1) use the numpy C-API to do this call faster
2) use range instead of np.arange (would end up being appreciably more invasive)

Comment (jreback) [reply pandas/_libs/internals.pyx]: @jbrockmendel absolutely (but non-cleanup PR)!

File: pandas/_libs/interval.pyx
Comment (jreback) [reply pandas/_libs/interval.pyx]: right, but that is orthogonal to this PR

File: pandas/_libs/tslib.pyx
Comment (jreback) [reply pandas/_libs/tslib.pyx]: can use astericks i think

**Review by jreback — CHANGES_REQUESTED** [review]
@MomIsBestFriend i think just some minor comments, anything that is mentioned as a perf change should be done independetnl. pls call out anything non-trivial (e.g. formatting) that is changed. or just don't do *anything* that is non formatting / style in this PR

File: pandas/_libs/tslib.pyx
Comment (WillAyd) [reply pandas/_libs/tslib.pyx]: I think should hold off on these types of changes until we get warnings to error, unless there is a known upside

Comment (jbrockmendel) [reply pandas/_libs/tslib.pyx]: its the usage encouraged by the cython folks, supposed to be slightly faster.  very marginal

**Review by WillAyd — APPROVED** [review]
lgtm as a style fixup; @jbrockmendel has valid comments but I think should all be done separate

**Comment by simonjayhawkins** [comment]
@MomIsBestFriend IIUC cython support for black is under development, https://github.com/psf/black/issues/359.

**Comment by MomIsBestFriend** [comment]
> @MomIsBestFriend IIUC cython support for black is under development, [psf/black#359](https://github.com/psf/black/issues/359).

Thank you @simonjayhawkins, me and my OCD are thrilled, I subscribed to that thread :laughing:

**Comment by WillAyd** [comment]
Can you merge master? Otherwise I think this is good to go

**Review by datapythonista — APPROVED** [review]
lgtm, nice clean up @MomIsBestFriend 

If you address the review comments, please do it in a separate PR. This touches a lot of code, better not to have any functional change here, only style changes.

Thanks!

**Comment by simonjayhawkins** [comment]
@MomIsBestFriend looking at the diff of the merge master, it looks like you've lost some of your clean-ups.

**Comment by MomIsBestFriend** [comment]
> @MomIsBestFriend looking at the diff of the merge master, it looks like you've lost some of your clean-ups.

@simonjayhawkins Correct, in 1cc4e89 I have reverted everything lost in the merge conflicts.

**Review by simonjayhawkins — APPROVED** [review]
Thanks @MomIsBestFriend lgtm pending green

**Comment by datapythonista** [comment]
All green. @jreback you requested changes, if you want to have another look, this should be ready.

**Comment by jreback** [comment]
ok as long as purely cosmetic

**Comment by WillAyd** [comment]
Thanks @MomIsBestFriend

**Pull request merged**
